### ISSUE TITLE
Data Visualisation - showing our data

### ISSUE THREAD

**Issue by richpomfret** [FreeUKGen/FreeUKRegProductIssues#1039]
Are there any new ways of showing our genealogical data that would be useful for users? Perhaps we could invite innovative ways of showing our data in ways that might help i.e. maps/timelines?

**Comment by edickens** [comment]
If I have understood this correctly, it requires entries to be linked into families.  We took a decision many years ago not to do this.  Ancestry does it automatically and gets it wrong quite often.  Or it relies on the researcher entering links, and they are often not correct too.  We leave it too the researcher to create their own family tree using a genealogy package, and some of these do have maps and timellines.
BUT, we could show research results on a map or timeline so that the researcher could then click on the one they think may be who they are looking for.

**Comment by richpomfret** [comment]
Yes I agree - this should be limited to new ways of showing the data/results we already provide or indeed, providing results on a map/timeline as you mention.

**Comment by Sherlock21** [comment]
Please expand on exactly what you mean, Rich.
At first reading the story, I guessed that you might mean something like:-researcher names a County, and  is then presented with a map of that County showing blobs at the Place Name locations where we do have transcribed date. It would need to be zoom in able, and cater for those who use PC's rather than smart phones and so can't "zoom' in the same way.
If its this proposal, there is already one by myself from way back.
If its not, then I would need some more clues please before adding further comment

**Comment by richpomfret** [comment]
I was aware something like this had been proposed in the past but was unsure of the details, what you suggest is pretty spot-on to how I imagined it, Eric. We are looking to get a few new tech volunteers in as part of a coding initative (during summer) and this might be a nice project/feature for them to potentially work on. Could you email the proposal to me and indeed, any other similar ideas you might have? The criteria is quite open, we simply would like innovative/new ways of showing the data for users/researchers that might help them to make other connections/links.

**Comment by edickens** [comment]
What I think is more important is the way we show what has been transcribed.  It was suggested that we have a map of places showing those places where we have transcriptions.  And the size of the dot, or colour, should show how much we have done.
If you want an email to send to the programmers, perhaps copy and paste from this story.

**Comment by Sherlock21** [comment]
yes, eric D

OR can someone find my original "story' on Github please. I cannot do that.

As Eric D says: I was concerned only with showing WHERE ( i.e. Places) we had transcribed some of the records.
Whereas Rich keeps referring to: New ways of Showing  the data to users. Which is completely different in my understanding, and I have visions of it being suggested that  clicking on by coloured blob would just bring up a big screen full of all the 10,000 Transcription records that we have.
Please tell me that's NOT what you mean.

AS to the original idea of mine, Ben and I had it all worked out  by the time we got to the end of doing what you presently see - so I would not be surprised if he has still got it somewhere. I don't get to see the back stage workings, and I can't get to find the original idea in a Github story ask say elsewhere.To some extent it was a 'stand alone' thing with Kirk, Ben & Myself.

**Comment by PatReynolds** [comment]
Stories #48 #179 #180 and #419 cover various aspects of data visualisation.

**Comment by RohitLearner** [comment]
Hi everybody, I m Rohit. I m in extraordinary touch with Visual Outline and Information Portrayal. I make them conceptualize thought that I would love to talk about with you. For what reason to simply run with maps/ timelines when there are different better choices.

The following are my suggestions : 
1. Data visualization using decision trees and clustering:
Reference: http://ieeexplore.ieee.org/document/7294400/; Lingo Clustering Algorithm with various different representation.

2. Graph and Charts : (Area and Scatter Charts)
Reference: Falcon SQL Client ( We can change our data into SQL and apply it to modification for different appearances.

3.Various other methods ( Other Brainstorming ideas are also available in my brains 👍 )

Tell me if I m going in right direction and which genealogical datasets are needed to be represented. Link to that dataset is required in reply. 

Let me know whether I m going in the right way and which genealogical datasets are should have been spoken to. Connection to that dataset is required in the answer.

**Comment by abu-kurian** [comment]
Hi,
My name is Abu Kurian, I had seen the organization listed in GSOC. 
I feel the best way to represent the given data set would be in the form of a graph with nodes representing a person and edges representing their relation to other nodes.  
I had worked on a similar project where a genealogy tree was made for authors in the field of computer science.
I have also attached a screen shot of the computer science genealogy tree just to give an idea.
Hoping to hear back soon.
![Screenshot from 2017-08-20 13_59_54.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5a8dae2498292d0012c414bf/125516c66e82c728ace21e0d46db978826878dba87e6ab03a24fff99691477227e49f3733dd7b368ee11445456304ba853444d04bea6d37ef5f22e7a985908b8823c0a7dcf3919acbfd01abf03ba208a4cd0b6bfdc5edcfa7f74b9dfd0126754505a9bf54f34a7934acd59ff49e24852ac113663e4647b.png)

**Comment by richpomfret** [comment]
Thanks for your suggestions. We are still in the process of fleshing out the requirements here and your ideas certainly help us. We will aim to provide more info soon! :)

**Comment by benwbrum** [comment]
See sample census data and corresponding explanation at https://github.com/FreeUKGen/SummerOfCodeImages/tree/master/freecen_subset/data

**Comment by benwbrum** [comment]
We're really talking about two different styles of visualization here:

* Record coverage visualization: this would allow researchers to see the records we possess for a given place and time.  The source records for both FreeREG and FreeCEN contain place coordinates, as well as dates for the records and vital dates (ages, etc) for the people mentioned within the data.  Displaying these as maps, timelines, and interactive visualizations would be invaluable.  Of secondary importance would be visualizations of distributions of surnames or forenames from the data -- this kind of thing tends to attract public interest.  Finally, visualizing mortality/natality statistics for counties or places over time might highlight events of interest to historians, such as outbreaks of disease or famine. 
* Search result visualization: This would allow users to see the records returned for their search on a map, letting them narrow in on specific areas by launching new searches for a place on the map.

**Comment by Sherlock21** [comment]
Given the mapping suggestion by Ben, does that have a bearing on our other story of changing the Genuki Url for each Placename in FreeREG? or just removing the link completely, as was discussed yesterday between Kirk & myself.

**Comment by PatReynolds** [comment]
I don't think so.  The search result visualisation would show 150 (or 15 or howevermany) on a map - so you could see that there was a geographical clustering.  The GENUKI feature just shows you where one place is, and (very usefully) opens up the GENUKI resources about that place to the user.

**Comment by Captainkirkdawson** [comment]
Mapping will be totally dependent on there being location information

It will be totally independent on our linking to genuki

If developed it might have to access genuki maps or use another source of
maps

Kirk Dawson
5220 Riverside Drive
Fairmont Hot Springs, B.C.
V0B 1L1

**Comment by shrutipabboju** [comment]
Hello,

I'm Shruti. I deal with data cleaning, transforming and visualization. I understand the need for Maps/Timelines instead of many other sources of Visualization. Although developers and people with already in depth knowledge of better visualization techniques would beg to differ, end users would find it gripping and easily understandable with Maps as the scheme of Visualization and I would like to contribute further on this.

With various parameters of Time, Year and other if provided with more Information could help with better visualization of maps or timelines.

Do tell me if I am going on the right path with this and other alternatives that we could develop!

Thanks :)

**Comment by edickens** [comment]
I've not got involved with this thread, so I may be suggesting something that is not relevant, but this is something I did some time ago to visualise the availability of entries in our "Transcriptions" option.  Known as the RAP, or Records Available Page.
![capture](https://user-images.githubusercontent.com/8876165/37894170-4164fe02-30d5-11e8-8f97-dd143045a4cd.JPG)

**Comment by smrr723** [comment]
This type of visualisation would need a fair bit of data, of which I think we have(?), but might be useful for visualising the gaps in records or volume of data we have for each region of the UK, using a slider to scroll through the years - [http://manpopex.us/](http://manpopex.us/)  - just throwing it out there because it looks great, and gets information across easily.

![visualisation](https://user-images.githubusercontent.com/2768445/39937328-44447d46-5547-11e8-98af-95a847b9366e.png)

**Comment by edickens** [comment]
Scott,
We do have this data.  We have the number of records by County, or Place, and we have the coordinates of the place.
But I ask - What would a researcher use this for?
Your example shows population density, which is necessary for planning services.

**Comment by smrr723** [comment]
Hello Eric - I was mostly just throwing this out there as another idea to generate discussion.

The above doesn't need to be limited to population density - you could make it so that the user can choose the variable.  

I'm not as familiar as others might be with the use cases researchers would need this for but I think it would be useful for viewing something like the change in distribution of a surname through the country over time, or change in average age of death across the country (if you were looking for outbreak of disease, or another potential cause of increased mortality, like the Icelandic volcanic ash I recently spoke to Ben about, that a researcher had asked for data on)... etc etc.

**Comment by richpomfret** [comment]
@benwbrum @edickens @Sherlock21 @Captainkirkdawson I am opening this up again for discussion as part of our next Google Summer of Code. It sounds like we could have two mini projects here (#419 might cover one of them).

**Comment by stsccfr** [comment]
Closing as old and beyond our means for the value it would add.  Reopen when our database becomes large enough that visualization is more useful.
